services:
  django_db:
    image: postgres:16
    container_name: django_db
    environment:
      POSTGRES_USER: "${DJANGO_POSTGRES_USER}"
      POSTGRES_PASSWORD: "${DJANGO_POSTGRES_PASSWORD}"
      POSTGRES_DB: "${DJANGO_POSTGRES_DB}"
    ports:
      - "5435:5432"
    volumes:
      - django_data:/var/lib/postgresql/data
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U ${DJANGO_POSTGRES_USER} -d ${DJANGO_POSTGRES_DB}" ]
      interval: 5s
      timeout: 5s
      retries: 5
    restart: unless-stopped

  django_backend:
    build:
      context: ./backend_django
      dockerfile: ./Dockerfile
    image: it-store-backend:latest
    container_name: it-store-backend
    volumes:
      - ./backend_django/mysite:/webapp/src
    environment:
      SECRET_KEY: "${DJANGO_SECRET_KEY}"
      DATABASE_URL: "postgres://${DJANGO_POSTGRES_USER}:${DJANGO_POSTGRES_PASSWORD}@django_db:5432/${DJANGO_POSTGRES_DB}"
      ALLOWED_HOSTS: "${DJANGO_ALLOWED_HOSTS}"
      DEBUG: "${DJANGO_DEBUG}"
    ports:
      - "8000:8000"
    depends_on:
      django_db:
        condition: service_healthy
    command:
      - sh
      - -c
      - |
        python manage.py migrate
        python manage.py runserver 0.0.0.0:8000


volumes:
  django_data: